inline_verifier.go

```diff
@@ -229,20 +230,49 @@ type InlineVerifier struct {
 	sourceStmtCache *StmtCache
 	targetStmtCache *StmtCache
 	logger          *logrus.Entry
+
+	// Used only for the ControlServer initiated VerifyDuringCutover
+	backgroundVerificationResultAndStatus VerificationResultAndStatus
+	backgroundVerificationErr             error
+	backgroundVerificationWg              *sync.WaitGroup
 }
 
+// This is called from the control server, which is triggered by pushing Run
+// Verification during cutover.
+// This step is necessary to ensure the binlogs are verified in Ghostferry.
 func (v *InlineVerifier) StartInBackground() error {
-	// not needed?
+	if v.verifyDuringCutoverStarted.Get() {
+		return errors.New("verification during cutover has already been started")
+	}
+
+	v.backgroundVerificationResultAndStatus = VerificationResultAndStatus{
+		StartTime: time.Now(),
+		DoneTime:  time.Time{},
+	}
+	v.backgroundVerificationErr = nil
+	v.backgroundVerificationWg = &sync.WaitGroup{}
+
+	v.logger.Info("starting InlineVerifier.VerifyDuringCutover in the background")
+
+	v.backgroundVerificationWg.Add(1)
+	go func() {
+		defer func() {
+			v.backgroundVerificationResultAndStatus.DoneTime = time.Now()
+			v.backgroundVerificationWg.Done()
+		}()
+
+		v.backgroundVerificationResultAndStatus.VerificationResult, v.backgroundVerificationErr = v.VerifyDuringCutover()
```

Comment:
```suggestion
		v.backgroundVerificationResultAndStatus.VerificationResult, v.backgroundVerificationErr = v.VerifyDuringCutover()
		v.verificationResultAndStatus.DoneTime = time.Now()
```

Do we not need to record the `DoneTime`? (I'm looking at the IterativeVerifier as a reference):

https://github.com/Shopify/ghostferry/blob/3113d9c4da8ac6e01b7ff75c9e852f11dcd58584/iterative_verifier.go#L274

Reply:
turns out this is set above in the `defer` 👍